taca_ngi_pipeline/deliver/deliver_dds.py

```diff
@@ -0,0 +1,545 @@
+
+    def delete_delivery_token_in_charon(self):
+        """Removes delivery_token from Charon upon successful delivery
+        """
+        charon_session = CharonSession()
+        charon_session.project_update(self.projectid, delivery_token='NO-TOKEN')
+
+    def add_dds_name_delivery_in_charon(self, name_of_delivery):
+        """Updates delivery_projects in Charon at project level
+        """
+        charon_session = CharonSession()
+        try:
+            #fetch the project
+            project_charon = charon_session.project_get(self.projectid)
+            delivery_projects = project_charon['delivery_projects']
+            if name_of_delivery not in delivery_projects:
+                delivery_projects.append(name_of_delivery)
+                charon_session.project_update(self.projectid, delivery_projects=delivery_projects)
+                logger.info('Charon delivery_projects for project {} '
+                            'updated with value {}'.format(self.projectid, name_of_delivery))
+            else:
+                logger.warn('Charon delivery_projects for project {} not updated '
+                            'with value {} because the value was already present'.format(self.projectid, name_of_delivery))
+        except Exception as e:
+            logger.exception('Failed to update delivery_projects in charon while '
+                         'delivering {}.'.format(self.projectid))
+
+    def add_dds_name_delivery_in_statusdb(self, name_of_delivery):
+        """Updates delivery_projects in StatusDB at project level
+        """
+        save_meta_info = getattr(self, 'save_meta_info', False)
+        if not save_meta_info:
+            return
+        status_db = ProjectSummaryConnection(self.config_statusdb)
+        project_page = status_db.get_entry(self.projectid, use_id_view=True)
+        delivery_projects = []
+        if 'delivery_projects' in project_page:
+            delivery_projects = project_page['delivery_projects']
+
+        delivery_projects.append(name_of_delivery)
+
+        project_page['delivery_projects'] = delivery_projects
+        try:
+            status_db.save_db_doc(project_page)
+            logger.info('Delivery_projects for project {} updated with value {} in statusdb'.format(self.projectid, name_of_delivery))
+        except Exception as e:
+            logger.exception('Failed to update delivery_projects in statusdb while delivering {}.'.format(self.projectid))
+
+    def upload_data(self, name_of_delivery):
+        """Upload staged sample data with DDS
+        """
+        stage_dir = self.expand_path(self.stagingpath)
+        log_dir = os.path.join(os.path.basename(CONFIG.get('log').get('file')), 'DDS_logs') 
+        project_log_dir = os.path.join(log_dir, self.projectid)
+        cmd = ['dds', 'data', 'put', 
+               '--mount-dir', project_log_dir,
+               '--project', name_of_delivery, 
+               '--source', stage_dir,
+               '--no-prompt']
```

Comment:
`--no-prompt` is an option of the dds command. I think it should be `dds --no-prompt data put ...`, but not completely sure.

Reply:
Whoops! Fixed 🙂